ODIN_II/SRC/include/BlockMemories.hpp

```diff
@@ -57,11 +58,28 @@ struct block_memory {
     char* memory_id;
 };
 
-typedef std::unordered_map<std::string, block_memory*> block_memory_hashtable;
+typedef std::unordered_map<std::string, block_memory_t*> block_memory_hashtable;
 extern block_memory_hashtable block_memories;
 
+/**
+ * block memories information. variable will be invalid
+ * after iterations happen before partial mapping
+*/
+struct block_memory_information_t {
+    /**
+     * block memory linked list, required for iterations 
+     * in optimization and mapping phase.
+    */
+    vtr::t_linked_vptr* block_memory_list;
```

Comment:
It would be good to comment exactly how to use each member variable.  E.g. first one is a linked list of pointers to some information for every block memory (not sure what information though -- should say what it points to!).  Or is the first linked list only for r/w memories?

Second one is the same, but for read-only memories?

Reply:
I have added more comments on how they are used, what they are including and when they will be invalid. I also briefly explained the concept behind BRAM/ROM naming convention that I used.